### PR TITLE
fix: recovery flow

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- With introducing the one-stop code flows we overwritten some of the recovery flow logic.
- This reintroduces/cleans this logic
- Verified locally all code flows work + all signups work

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/components/auth/CodeVerificationForm.tsx [hunk 1]
```diff
@@ -11,7 +11,6 @@ import { AuthEventNames } from '../../lib/auth';
 import AnalyticsContext from '../../contexts/AnalyticsContext';
 import AuthForm from './AuthForm';
```
Comment (nensidosari) [inline packages/shared/src/components/auth/CodeVerificationForm.tsx:12]: Just something that I found very confusing when reviewing the pervious and this PR..the file names 😭 

EmailCodeVerification, CodeVerificationForm

**Review by nensidosari — APPROVED** [review]
Looks good

File: packages/shared/src/components/auth/CodeVerificationForm.tsx [hunk 2]
```diff
@@ -32,17 +31,14 @@ function CodeVerificationForm({
...
       flowId: initialFlow,
       onTimerFinished: () => setEmailSent(false),
       onError: setHint,
       onSuccess: () => {
         setEmailSent(true);
       },
       onVerifyCodeSuccess: () => {
-        trackEvent({
-          event_name: AuthEventNames.VerifiedSuccessfully,
-        });
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/auth/CodeVerificationForm.tsx:45]: Yep, was accidentally added thinking it was the other flow 😅

File: packages/shared/src/components/auth/CodeVerificationForm.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/auth/CodeVerificationForm.tsx:12]: I know! Sole reason this was introduced wrongly 🤯
Really want to prio some work on auth to abstract/redo some stuff and fix naming/imports.

**Pull request merged**